File: plugins/memory/memory.10s.js
#!/usr/bin/env node
/**
 * Memory Monitor Plugin - Uses Crossbar API for portability
 */
const { execSync } = require('child_process');
const os = require('os');

function crossbar(cmd) {
    try {
        return execSync(`crossbar ${cmd}`, { encoding: 'utf8', timeout: 5000 }).trim();
    } catch {
        return null;
    }
}

let memoryStr = crossbar('--memory');

if (!memoryStr) {
    const total = os.totalmem();
    const free = os.freemem();
    const percent = Math.round(((total - free) / total) * 100);
    memoryStr = percent.toString();
}

const memory = parseInt(memoryStr) || 0;
let color = 'green';
if (memory > 80) color = 'red';
else if (memory > 60) color = 'yellow';

console.log(`🧠 ${memoryStr}% | color=${color}`);
console.log('---');
console.log(`Memory Usage: ${memoryStr}%`);
console.log('---');
console.log('Refresh | refresh=true');

File: plugins/memory/memory.10s.rs
//! Memory Monitor Plugin - Uses Crossbar API for portability

use std::process::Command;

fn crossbar(args: &str) -> Option<String> {
    let output = Command::new("crossbar")
        .args(args.split_whitespace())
        .output()
        .ok()?;
    
    if output.status.success() {
        Some(String::from_utf8_lossy(&output.stdout).trim().to_string())
    } else {
        None
    }
}

fn main() {
    let memory_str = crossbar("--memory").unwrap_or_else(|| "N/A".to_string());
    
    let color = if let Ok(memory) = memory_str.trim_end_matches('%').parse::<i32>() {
        if memory > 80 { "red" }
        else if memory > 60 { "yellow" }
        else { "green" }
    } else {
        "gray"
    };

    println!("🧠 {}% | color={}", memory_str, color);
    println!("---");
    println!("Memory Usage: {}%", memory_str);
    println!("---");
    println!("Refresh | refresh=true");
}
